Jenkinsfile

```diff
@@ -1 +1,4 @@
-buildPlugin(platforms: ['docker', 'windows'])
+buildPlugin(useContainerAgent: true, configurations: [
+  [platform: 'docker', jdk: 21],
+  [platform: 'windows', jdk: 17],
```

Comment:
Trying to use a VM for Linux builds (so we will run some Dockerized tests when we can) and a container for Windows builds (skipping Dockerized tests). Does not seem to be directly supported as of https://github.com/jenkins-infra/pipeline-library/pull/522 unless I am missing something @smerle33 @dduportal. https://github.com/jenkins-infra/pipeline-library/pull/658 is maybe not quite for this case.